arrows/ffmpeg/ffmpeg_init.cxx

```diff
@@ -35,22 +35,70 @@
 
 #include "ffmpeg_init.h"
 
+
+#include <algorithm>
+#include <cctype>
 #include <mutex>
 
 extern "C" {
 #include <libavformat/avformat.h>
 }
 
+#include <vital/logger/logger.h>
+
 std::mutex ffmpeg_init_mutex;
 
-//--------------------------------------------------------------------------------
+static auto ffmpeg_logger = kwiver::vital::get_logger( "arrows.ffmpeg" );
+
+//-----------------------------------------------------------------------------
+// trim whitespace from end of string (in place)
+static inline void rtrim(std::string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(),
+                         [](int ch) { return !std::isspace(ch);}
+                         ).base(), s.end());
+}
+
+//-----------------------------------------------------------------------------
+void
+ffmpeg_kwiver_log_callback(void* ptr, int level, const char* fmt, va_list vl)
+{
+  static int print_prefix = 0;
+  char line[1024];
+  av_log_format_line(ptr, level, fmt, vl, line, sizeof(line), &print_prefix);
+  std::string msg(line);
+  rtrim(msg);
+  switch(level)
+  {
+    case AV_LOG_PANIC:
+    case AV_LOG_FATAL:
+      LOG_ERROR(ffmpeg_logger, msg);
+      break;
+    case AV_LOG_ERROR:
+    case AV_LOG_WARNING:
+      LOG_WARN(ffmpeg_logger, msg);
+      break;
+    case AV_LOG_INFO:
+      LOG_INFO(ffmpeg_logger, msg);
+      break;
+    case AV_LOG_VERBOSE:
+    case AV_LOG_DEBUG:
+      LOG_DEBUG(ffmpeg_logger, msg);
+      break;
+    default:
+      break;
+  };
+
+}
+
+//-----------------------------------------------------------------------------
 void ffmpeg_init()
 {
   std::lock_guard< std::mutex > lock(ffmpeg_init_mutex);
   static bool initialized = false;
   if ( ! initialized ) {
     av_register_all();
     av_log_set_level(AV_LOG_ERROR);
```

Comment:
I assume that this limits logging to ERRORS and above. This is also appears to be done by the logging properties file. It may be useful, at some point, to see more than error logs generated by FFMPEG.

Reply:
That was left over from before.  I'm actually not sure if it does anything when not using the ffmpeg default logger.  During testing I was seeing message from FFMPEG at various different levels (warnings, info, etc.)  maybe I should just remove that line?

Reply:
If it doesn't seem to have any effect, then you could remove it.

Reply:
I'll remove it and test again